tests/test_rule_validation.py

```diff
@@ -0,0 +1,125 @@
+import unittest
+
+import app.validation.validator as validation
+
+single_answer_id = {
+    "confirm-zero-employees-answer": {
+        "answer": {
+            "type": "Radio",
+            "id": "confirm-zero-employees-answer",
+            "options": [
+                {"label": "Yes this is correct", "value": "Yes this is correct"},
+                {
+                    "label": "No I need to change this",
+                    "value": "No I need to change this",
+                },
+            ],
+            "mandatory": True,
+            "q_code": "d50",
+        },
+        "block": "confirm-zero-employees-block",
+        "group_id": "confirmation-block",
+        "section": "default-section",
+    }
+}
+
+multiple_answer_ids = {
+    "accommodation-type-answer": {
+        "answer": {
+            "id": "accommodation-type-answer",
+            "mandatory": True,
+            "options": [
+                {
+                    "description": "For example, student hall of residence, boarding school, "
+                    "armed forces base, hospital, care home, prison",
+                    "label": "A communal establishment",
+                    "value": "A communal establishment",
+                },
+                {
+                    "label": "A private or family household",
+                    "value": "A private or family household",
+                },
+            ],
+            "type": "Radio",
+        },
+        "block": "accommodation-type",
+        "group_id": "personal-details-group",
+        "section": "individual-section",
+    },
+    "proxy-answer": {
+        "answer": {
+            "default": "Yes",
+            "id": "proxy-answer",
+            "mandatory": False,
+            "options": [
+                {
+                    "label": "No, I’m answering for myself",
+                    "value": "No, I’m answering for myself",
+                },
+                {"label": "Yes", "value": "Yes"},
+            ],
+            "type": "Radio",
+        },
+        "block": "proxy",
+        "group_id": "personal-details-group",
+        "section": "individual-section",
+    },
+}
+
+
+class TestRule(unittest.TestCase):
+    @staticmethod
+    def test_search_first_value_in_options():
+
+        comparison = validation.Validator().is_rule_value_valid(
+            single_answer_id, "Yes this is correct"
+        )
+
+        assert comparison is True
+
+    @staticmethod
+    def test_search_second_value_in_options():
+
+        comparison = validation.Validator().is_rule_value_valid(
+            single_answer_id, "No I need to change this"
+        )
+
+        assert comparison is True
+
+    @staticmethod
+    def test_number_rule_ignored():
+
+        comparison = validation.Validator().is_rule_value_valid(single_answer_id, 123)
```

Comment:
Shouldn't this return false? The value isn't possible to enter with the rule given. The test also seems to be inappropriately named, as the return value implies it hasn't been ignored.

Reply:
If this returned false it would not find an answer and continue searching through answers map. Integer routing values are ignored because tracking them in answers map is problematic(they don't necessarily need to exist in answer values to work as routing rules).

Reply:
This role of this method isn't to locate an answer, it's to test if a given answer is valid against the rule that defines it. If you need to allow other values, you should add further methods to allow for this.

At the moment this reads as the method confirms is_rule_valid(some_map_of_valid_values, invalid_value), which isn't correct.

Reply:
But in some instances, integer rule value cannot be valid against any answer in schema, yet still be valid. Validator isn't able to check that because such answer value doesn't exist. This test checks if integer rule is being ignored as expected.

Reply:
This is addressed now, thanks!